kms_log/main.tf

```diff
@@ -558,4 +593,469 @@ resource "aws_cloudwatch_metric_alarm" "dead_letter" {
+
+        resources = [
+            "${aws_kms_key.kms_logging.arn}"
+        ]
+    }
+}
+
+data "aws_iam_policy_document" "cwprocessor_sns" {
+    statement {
+        sid = "SNS"
+        effect = "Allow"
+        actions = [
+            "sns:Publish"
+        ]
+
+        resources = [
+            "${aws_sns_topic.kms_logging_events.arn}"
+        ]
+    }
+}
+
+data "aws_iam_policy_document" "cwprocessor_kinesis" {
+    statement {
+        sid = "Kinesis"
+        effect = "Allow"
+        actions = [
+            "kinesis:GetShardIterator",
+            "kinesis:GetRecords",
+            "kinesis:DescribeStream"
+        ]
+
+        resources = [
+            "${aws_kinesis_stream.datastream.arn}"
+        ]
+    }
+}
+
+resource "aws_iam_role_policy" "cwprocessor_cloudwatch" {
+    name = "CloudWatch"
+    role = "${aws_iam_role.cloudwatch_processor.id}"
+    policy = "${data.aws_iam_policy_document.ctprocessor_cloudwatch.json}"
+}
+
+resource "aws_iam_role_policy" "cwprocessor_dynamodb" {
+    name = "DynamoDb"
+    role = "${aws_iam_role.cloudwatch_processor.id}"
+    policy = "${data.aws_iam_policy_document.cwprocessor_dynamodb.json}"
+}
+
+resource "aws_iam_role_policy" "cwprocessor_kms" {
+    name = "KMS"
+    role = "${aws_iam_role.cloudwatch_processor.id}"
+    policy = "${data.aws_iam_policy_document.cwprocessor_kms.json}"
+}
+
+resource "aws_iam_role_policy" "cwprocessor_sns" {
+    name = "KMS"
+    role = "${aws_iam_role.cloudwatch_processor.id}"
+    policy = "${data.aws_iam_policy_document.cwprocessor_sns.json}"
+}
+
+resource "aws_iam_role_policy" "cwprocessor_kinesis" {
+    name = "Kinesis"
+    role = "${aws_iam_role.cloudwatch_processor.id}"
+    policy = "${data.aws_iam_policy_document.cwprocessor_kinesis.json}"
+}
+
+# lambda for creating cloudwatch metrics and events
+resource "aws_lambda_function" "event_processor" {
+    count = "${var.kmslogging_service_enabled}"
+    s3_bucket = "${data.aws_s3_bucket.lambda.id}"
+    s3_key = "circleci/identity-lambda-functions/${var.lambda_identity_lambda_functions_gitrev}.zip"
+
+    lifecycle {
+        ignore_changes = ["s3_key", "last_modified"]
+    }
+
+    function_name = "${var.env_name}-kmslog-event-processor"
+    description = "18F/identity-lambda-functions: KMS Log Event Processor"
+    role = "${aws_iam_role.event_processor.arn}"
+    handler = "main.Functions::KMSEventHandler.process"  #TODO Verify
+    runtime = "ruby2.5"
+    timeout = 30 # seconds
+
+    environment {
+        variables = {
+            DEBUG = "${var.lambda_audit_github_debug ? "1" : ""}"
+            LOG_LEVEL = "0"
+            ENV_NAME = "${var.env_name}"
+        }
+    }
+
+    tags {
+        source_repo = "https://github.com/18F/identity-lambda-functions"
+        environment = "${var.env_name}"
+    }
+}
+
+resource "aws_lambda_event_source_mapping" "event_processor" {
+    count = "${var.kmslogging_service_enabled}"
+    event_source_arn = "${aws_sqs_queue.kms_cw_events.arn}"
```

Comment:
This doesn't seem to be defined. Is it the same as kms_cloudwatch_events?

Reply:
It is a 3rd lambda for sending data to cloudwatch events.

Reply:
I mean that the SQS queue doesn't seem to be defined in this file. Terraform gives me an error.

Reply:
The queue was getting created, the reference wasn't correct.  I have updated it.